### PR TITLE
AVRO-3585: Add support for encoding Value::String as Schema::Uuid

### PR DESCRIPTION
**Pull request by RikHeijdens**
Make sure you have checked _all_ steps below.

### Jira

- [x] https://issues.apache.org/jira/browse/AVRO-3585

### Tests

- [x] My PR adds the following unit tests

### Commits

- [x] My commits all reference Jira issues in their subject lines. In addition, my commits follow the guidelines from "[How to write a good git commit message](https://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [X] No need

### CODE REVIEW

File: lang/rust/avro/src/encode.rs [hunk 1]
```diff
@@ -134,6 +134,9 @@ pub(crate) fn encode_internal<S: Borrow<Schema>>(
                     return Err(Error::GetEnumSymbol(s.clone()));
                 }
             }
+            Schema::Uuid => {
```
Comment (jklamer) [inline lang/rust/avro/src/encode.rs:137]: You should be able to add this condition to the above. L 126

```
Schema::String | Schema::Uuid => ....
```

**Comment by jklamer** [comment]
It looks like the validation already allows this case: https://github.com/apache/avro/blob/master/lang/rust/avro/src/types.rs#L403 so regardless if others agree on the adding support we need to make the two consistent

File: lang/rust/avro/src/encode.rs (hunk 1 above)
Comment (martin-g) [reply lang/rust/avro/src/encode.rs:137]: Agreed!
@RikHeijdens Please update the PR with @jklamer's suggestion.

**Review by martin-g — CHANGES_REQUESTED** [review]

File: lang/rust/avro/src/encode.rs (hunk 1 above)
Comment (RikHeijdens) [reply lang/rust/avro/src/encode.rs]: Updated the PR with @jklamer's suggestion in https://github.com/apache/avro/pull/1778/commits/f0203f0ff4b46bb008c5c89d6856abc45c10cf3b!

**Review by martin-g — APPROVED** [review]

**Comment by martin-g** [comment]
Thank you for the contribution, @RikHeijdens !

Thank you for the review, @jklamer !

**Pull request merged**